homeassistant/components/ecobee/climate.py

```diff
@@ -330,6 +335,23 @@ async def async_update(self):
         if self.hvac_mode != HVAC_MODE_OFF:
             self._last_active_hvac_mode = self.hvac_mode
 
+        await self.throttled_validate_data()
+
+    @Throttle(VALIDATE_DATA_INTERVAL)
+    async def throttled_validate_data(self):
+        """Validate data from ecobee periodically and log errors."""
+        # Check that this is a known model, log message if not
+        if not self.thermostat["modelNumber"] in ECOBEE_MODEL_TO_NAME:
```

Comment:
I suggest we move this check to just before we create the entity. Make a for loop in `async_setup_entry`.
```py
entities = []

for index in range(len(data.ecobee.thermostats)):
    thermostat = data.ecobee.get_thermostat(index)
    if not thermostat["modelNumber"] in ECOBEE_MODEL_TO_NAME:
        ...
    entities.append(Thermostat(data, index, thermostat))
```

Reply:
Thanks. The check has been moved into `async_setup_entry`. I also needed to clean up the references to `device` in some of the subsequent service definitions.